sdk/cognitiveservices/azopenai/custom_client_image.go

```diff
@@ -0,0 +1,84 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azopenai
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// CreateImageOptions contains the optional parameters for the Client.CreateImage method.
+type CreateImageOptions struct {
+	// placeholder for future optional parameters
+}
+
+// CreateImageResponse contains the response from method Client.CreateImage.
+type CreateImageResponse struct {
+	ImageGenerations
+}
+
+// CreateImage creates an image using the Dall-E API.
+func (client *Client) CreateImage(ctx context.Context, body ImageGenerationOptions, options *CreateImageOptions) (CreateImageResponse, error) {
```

Comment:
A new nits here (can be fixed later).
`ImageGenerationOptions` needs a better name, maybe something like `ImageGenerationSettings` (or perhaps `CreateImageSettings` so it aligns with `CreateImageOptions`).  And ideally, it would contain all required settings and the optional ones moved to `CreateImageOptions`.

Reply:
Filed: https://github.com/Azure/azure-sdk-for-go/issues/21106

We should match, on data type name, with the other SDKs as much as possible.

Reply:
I mostly agree, unless the other SDKs got it pretty wrong.

Reply:
There's a discussion about it although the context of that was whether we wanted to make parameters positional or not - the main issue there being that there's quite a few. Splitting into options and non-options feels like a good split but I think the other languages have different criteria on whether "things named options" only contain optional parameters.